src/platform-includes/sourcemaps/upload/cli/javascript.mdx

```diff
@@ -1,53 +1,163 @@
+
+Once your artifact files are injected, upload them to Sentry with the `upload` command:
+
+```bash
+sentry-cli sourcemaps upload /path/to/file
+```
+
+You can verify your uploaded artifacts by navigating to **Project Settings -> Source Maps (left sidebar) -> Debug ID Bundles** on Sentry.
+
+At this point, the `sentry-cli` will notify you if there are files that don't contain Debug IDs (because for example, you forgot to run the `inject` command). The CLI will then prompt you to create a _Weak Release Association_ to connect a release to this artifact. This connection allows Sentry to properly symbolicate events without Debug IDs.
+
+#### Weak Release Associations
+
+_Weak Release Associations_ are an extension of the Debug ID system. By weakly associating artifacts with a `release`, and optionally a `dist`, Sentry can look up sources for symbolication, even if Debug IDs are missing.
+
+Creating a Weak Release Association doesn't create a release in Sentry. You must create a release with the same name you specified in the Weak Release Association as a separate step in your pipeline.
+
+It's important that the name of the `release` you specify during upload exactly matches the name of the `release` you create with the `sentry-cli` and/or the `release` in your SDK, e.g.:
 ```javascript
 Sentry.init({
-  // Make sure this value is identical to the name you give the release that you
-  // create below using Sentry CLI
+  // This value must be identical to the release name specified during upload with the `sentry-cli`
   release: "<release_name>",
 });
 ```
 
-## 2. Upload Source Maps Using Sentry CLI
+Add these parameters to the `upload` command to weakly associate artifacts with a `release` and optionally `dist`:
 
-<Note>
+```bash
+sentry-cli sourcemaps upload --release=<release_name> --dist=<dist_name> /path/to/file
+```
 
-You can find installation instructions for Sentry CLI here: https://docs.sentry.io/product/cli/installation/
+<Note>
 
-For more info on Sentry CLI configuration visit the [Sentry CLI configuration docs](/product/cli/configuration/).
+While it's possible to only use Weak Release Associations and skip Debug ID injection via the `inject` command, we strongly recommend using Debug ID injection if it works with your project setup.
 
 </Note>
 
-Make sure Sentry CLI is configured for your project. You can use a `.sentryclirc` file or environment variables:
+## Option 2: Upload with Release (Legacy)
+
+Sentry can also use [releases](/product/releases/) to match uploaded source maps to events. You can use `sentry-cli` to create releases and upload their source maps to Sentry.
 
 <Note>
 
-Visit the [auth token user settings page](https://sentry.io/settings/account/api/auth-tokens/) to find your auth tokens.
+We strongly suggest to use the new Debug IDs system for uploading your artifacts since it provides more reliable symbolication and the old release-based system will be faded out in the future.
 
 </Note>
```

Comment:
Instead of only saying that we recommend the new version, we should just say in which cases the legacy upload is useful. eg:

- bundler plugins don't work for some reason
- bundler plugins make bundle size too large
- integrity issues
- ...

> the old release-based system will be faded out in the future

This is not correct I think. We cannot phase out the old system. There are cases where debug id will not work. We shouldn't unnecessarily scare users with deprecation notices.

Reply:
We are already using an updated fetching system that doesn't use debug ids, thus everything will use the new endpoint and logic, irrespective of debug ids. The plan is to fade out the old system and use the weakly associated system or debug id one based on the user information.